File: backend/src/domain/password.ts
export default class Password {
    constructor(readonly value: string) {
        if (!this.validLength(value)) throw new Error("Invalid length, password must contain at least 6 characters")
        if (!this.existSpecialCharacter(value)) throw new Error("Password must contain at least one special character")
        if (!this.containACapitalLetter(value)) throw new Error("Must contain a capital letter")
        if (!this.containALowercaseLetter(value)) throw new Error("Must contain a lowercase letter")
        if (!this.containANumber(value)) throw new Error("Must contain numbers")
    }

    private validLength(password: string) {
        if (password.length < 6) return false
        return true
    }

    private existSpecialCharacter(password: string) {
        const regex = /.*[@!#$%^&*()/\\]/
        const patt = new RegExp(regex)
        return patt.test(password)
    }

    private containACapitalLetter(password: string) {
        const regex = /[A-Z]/
        const patt = new RegExp(regex)
        return patt.test(password)
    }

    private containALowercaseLetter(password: string) {
        const regex = /[a-z]/
        const patt = new RegExp(regex)
        return patt.test(password)
    }

    private containANumber(password: string) {
        const regex = /[\d]/
        const patt = new RegExp(regex)
        return patt.test(password)
    }
}

File: backend/auth/auth/domain/entities/password.cs
using System.Text.RegularExpressions;
public class Password {
    public string _value {get;}
    public Password(string value){
        if (!this.ValidLength(value))throw new ArgumentException("Invalid length, password must contain at least 6 characters");
        if (!ExistSpecialCharacter(value)) throw new ArgumentException("Password must contain at least one special character");
        if (!ContainACapitalLetter(value)) throw new ArgumentException("Must contain a capital letter");
        if (!ContainALowercaseLetter(value)) throw new ArgumentException("Must contain a lowercase letter");
        if (!ContainANumber(value)) throw new ArgumentException("Must contain numbers");
        this._value = value;
    }


     private bool ValidLength(string value) {
        if (value.Length < 6) return false;
        return true;
    }

    private bool ExistSpecialCharacter(string value) {
        string regexExpression = @".*[@!#$%^&*()/\\]";
        Regex regex = new Regex(regexExpression);
        return regex.IsMatch(value);
    }

    private bool ContainACapitalLetter(string value) {
        string regexexpression = "[A-Z]";
        Regex regex = new Regex(regexexpression);
        return regex.IsMatch(value);

    }

    private bool ContainALowercaseLetter(string value) {
        string regexexpression = "[a-z]";
        Regex regex = new Regex(regexexpression);
        return regex.IsMatch(value);
    }

    private bool ContainANumber(string value) {
        string regexexpression = @"[\d]";
        Regex regex = new Regex(regexexpression);
        return regex.IsMatch(value);
    }
}